src/transformers/models/chameleon/modeling_chameleon.py

```diff
@@ -1225,12 +1395,82 @@ def get_image_tokens(self, pixel_values: torch.FloatTensor):
         Args:
             pixel_values (`torch.FloatTensor` of shape `(batch_size, num_channels, image_size, image_size)):
                 The tensors corresponding to the input images.
+
+        Returns:
+            `torch.Tensor` of shape `(batch_size, vqmodel.quantize.quant_state_flattened_dim)`:
+                The BPE tokens generated by the model.
         """
-        batch_size = pixel_values.shape[0]
         _, _, image_toks = self.vqmodel.encode(pixel_values)
-        bpe_toks = self.vocabulary_mapping.convert_img2bpe(image_toks)
-        bpe_toks = bpe_toks.view(batch_size, -1)
-        return bpe_toks
+        return self.vocabulary_mapping.convert_img2bpe(image_toks)
+
+    def decode_image_tokens(self, bpe_tokens: torch.Tensor):
+        """
+        Converts BPE tokens generated by the model into discrete image tokens
+        compatible with the VQGAN module, then decodes them into pixel values.
+
+        Args:
+            bpe_tokens (`torch.tensor` of shape `(batch, vqmodel.quantize.quant_state_flattened_dim)`):
+                The BPE tokens generated by the model.
+
+        Returns:
+            `torch.Tensor` of shape `(batch, num_channels, 512, 512)`:
+        """
+        if not torch.is_tensor(bpe_tokens):
+            bpe_tokens = torch.tensor(bpe_tokens, device=self.device)
```

Comment:
Can we expect a non-tensor input? I'd say we should assume it's tensor, same as forward of models expects a tensor

Reply:
removed!